File: CSharp/leetcode/check-if-all-1s-are-at-least-length-k-places-away/solution.cs
public class Solution {
    public bool KLengthApart(int[] nums, int k) {
        int prev = -1;
        for (int i = 0; i < nums.Length; i++) {
            if (nums[i] == 1) {
                if (prev != -1 && i - prev - 1 < k) {
                    return false;
                }
                prev = i;
            }
        }
        return true;
    }
}

File: PHP/leetcode/check-if-all-1s-are-at-least-length-k-places-away/solution.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $k
     * @return Boolean
     */
    function kLengthApart($nums, $k) {
        $prev = -1;
        $n = count($nums);
        for ($i = 0; $i < $n; $i++) {
            if ($nums[$i] == 1) {
                if ($prev != -1 && $i - $prev - 1 < $k) {
                    return false;
                }
                $prev = $i;
            }
        }
        return true;
    }
}
